packages/language-server/src/plugins/typescript/features/FindReferencesProvider.ts

```diff
@@ -42,6 +42,10 @@ export class FindReferencesProviderImpl implements FindReferencesProvider {
                     );
                 })
         );
+
+        const nonZeroLocations = locations.filter(hasNonZeroRange);
+
+        return nonZeroLocations.length > 0 ? nonZeroLocations : locations;
```

Comment:
What's the reason for returning locations in case there's no non-zero range found?

Reply:
A bias toward preserving existing behaviour in the face of uncertainty, I suppose. But it might be misguided in this case...

The case I had vaguely in mind was if it returned references in another file, it might still be somewhat useful to jump to this file, even if no actual precise location can be found. That said, even then, it may do more harm than good, maybe giving the impression that the language tools is a bit lost, jumping to the middle of a file for no obvious reason. The behaviour would make much more sense for definitions and I think I got a bit confused.

I'm removing that.